lib/puppet/ssl/host.rb

```diff
@@ -371,27 +382,67 @@ def state
 
   private
 
+  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  def ensure_crl_if_needed
+    if use_crl? && !Puppet::FileSystem.exist?(crl_path)
+      # The CertificateRevocationList indirector will attempt to download the
+      # CRL from the CA if it does not exist on disk. It will ask host for
+      # its ssl_store again, but expect crl checking to be disabled for that
+      # store.
+      old_crl_usage = @crl_usage
+      @crl_usage = false
+      Puppet.debug _("Disabling certificate revocation checking when fetching the CRL and no CRL is present")
+      if !CertificateRevocationList.indirection.find(CA_NAME)
+        raise Puppet::Error,
+              _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
+      end
+      @crl_usage = old_crl_usage
+    end
+  end
+
+  # @param path [String] Path to CRL Chain
+  # @return [Array<OpenSSL::X509::CRL>] CRLs from chain
+  # @raise [Errno::ENOENT] if file does not exist
+  # @raise [Puppet::Error<OpenSSL::X509::CRLError>] if the CRL chain is malformed
+  def load_crls(path)
+    ending = "-----END X509 CRL-----\n"
+    crls_pems = Puppet::FileSystem.read(path, encoding: Encoding::ASCII)
+    crls_pems.split(ending).map do |crl|
+      begin
+        crl += ending
```

Comment:
In my PR I'm splitting on a look-behind regex `/(?<=#{ending})/` so I keep my splitter as part of the string. Not sure which of these is preferred, but I do think we should be consistent.

Reply:
Good call on the look behind. I was thinking about updating mine to use String#scan with a full capture based on the cert loading in SSL::Configuration. Do you have a preference between String#scan and the look-behind regex?